pymeasure/instruments/feeltech/fy3200s.py

```diff
@@ -0,0 +1,314 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, Channel
+from pymeasure.instruments.validators import strict_discrete_range, strict_discrete_set
+
+
+def step_validator(step):
+    """Creates a strict_discrete_range with the specified step"""
+    return lambda x, y: strict_discrete_range(x, y, step)
+
+
+class MainChannel(Channel):
+    WAVEFORMS = {
+        "SINE": 0,
+        "SQUARE": 1,
+        "PULSE": 2,
+        "TRIANGLE": 3,
+        "SAWTOOTH_RISE": 4,
+        "SAWTOOTH_FALL": 5,
+        "DC": 6,
+        "PULSE_LORENTZ": 7,
+        "MULTITONE": 8,
+        "RANDOM_NOISE": 9,
+        "ELECTROCARDIOGRAM": 10,
+        "PULSE_TRAPEZ": 11,
+        "PULSE_SINC": 12,
+        "PULSE_NARROW": 13,
+        "GAUSSIAN_NOISE": 14,
+        "AMP_MOD": 15,
+        "FREQ_MOD": 16,
+        "ARB1": 17,
+        "ARB2": 18,
+        "ARB3": 19,
+    }
+
+    SWEEPMODES = {"LINEAR": 0, "LOG": 1}
+
+    waveform = Instrument.setting(
+        "{ch}w%i",
+        """Set the waveform of the channel.
+
+        The waveform must be one of the following:
+
+        - SINE
+        - SQUARE
+        - PULSE
+        - TRIANGLE
+        - SAWTOOTH_RISE
+        - SAWTOOTH_FALL
+        - DC
+        - PULSE_LORENTZ
+        - MULTITONE
+        - RANDOM_NOISE
+        - ELECTROCARDIOGRAM
+        - PULSE_TRAPEZ
+        - PULSE_SINC
+        - PULSE_NARROW
+        - GAUSSIAN_NOISE
+        - AMP_MOD
+        - FREQ_MOD
+        - ARB1
+        - ARB2
+        - ARB3
+        """,
+        validator=strict_discrete_set,
+        map_values=True,
+        values=WAVEFORMS,
+    )
+
+    frequency = Instrument.control(
+        "{ch}\ncf",
+        "{ch}f%.9i",
+        "Control the frequency (Hz) of the channel.",
+        validator=step_validator(1),
+        values=(0, 100000000),
+        set_process=lambda x: x * 100,
+        get_process=lambda x: float(x.strip("cf")) / 100,
+    )
+
+    amplitude = Instrument.setting(
+        "{ch}a%.1f",
+        "Set the amplitude (V) of the channel.",
+        validator=step_validator(0.1),
+        values=(0.1, 20),
+    )
+
+    offset = Instrument.setting(
+        "{ch}o%.1f",
+        "Set the offset (V) of the channel",
+        validator=step_validator(0.1),
+        values=(-10, 10),
+    )
+
+    duty_cycle = Instrument.control(
+        "{ch}\ncd",
+        "{ch}d%i",
+        "Set the duty cycle (percentage) of the channel",
```

Comment:
You have a `control`, so please use also "Control", which indicates getting and setting:
```suggestion
        "Control the duty cycle (percentage) of the channel",
```